listener/listener.go

```diff
@@ -300,80 +315,150 @@ func (l *Listener) Stream(ctx context.Context) error {
 		},
 	}
 
-	tx := NewWalTransaction(l.log, pool, l.monitor)
-
-	for {
-		if err := ctx.Err(); err != nil {
-			l.log.Warn("stream: context canceled", "err", err)
-			return nil
-		}
-
-		msg, err := l.replicator.WaitForReplicationMessage(ctx)
-		if err != nil {
-			return fmt.Errorf("wait for replication message: %w", err)
-		}
-
-		if msg == nil {
-			l.log.Debug("got empty message")
-			continue
-		}
-
-		if err = l.processMessage(ctx, msg, tx); err != nil {
-			return fmt.Errorf("process message: %w", err)
+	topicSet := make(map[string]bool)
+	defer func() {
+		for topic := range topicSet {
+			l.publisher.Flush(topic)
```

Comment:
Upstream flushes after every message and main on our fork doesn't flush, figured we just want to on teardown. Otherwise it should handle it's own flushing every 8MB.